### PR TITLE
Update test_plan.py to get token via az cli

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
This is the second part of the work for eliminating client secrets.

This change depends on merge of #13435.

#### How did you do it?
This change updated the test_plan.py to get access token by az cli. For the az cli command to work, the task calling test_plan.py must be wrapped in an AzureCLI azure pipeline task.

### CODE REVIEW

**Review by lerry-lee — APPROVED** [review]
LGTM

**Review by StormLiangMS — APPROVED** [review]
approved with a question

File: .azure-pipelines/test_plan.py [hunk 1]
```diff
@@ -130,16 +131,16 @@ def __init__(self):
...
-    scope = "api://sonic-testbed-tools-dev/.default"
-    if elastictest_url in [
-        "http://sonic-testbed2-scheduler-backend.azurewebsites.net",
-        "https://sonic-testbed2-scheduler-backend.azurewebsites.net",
-        "http://sonic-elastictest-prod-scheduler-backend-webapp.azurewebsites.net",
-        "https://sonic-elastictest-prod-scheduler-backend-webapp.azurewebsites.net"
-    ]:
-        scope = "api://sonic-testbed-tools-prod/.default"
-    return scope
+# def get_scope(elastictest_url):
```
Comment (StormLiangMS) [inline .azure-pipelines/test_plan.py:134]: should we remove this code other than keep them in the source code with comment out?

**Review by StormLiangMS — APPROVED** [review]
approved with a question.

File: .azure-pipelines/test_plan.py (hunk 1 above)
Comment (wangxin) [reply .azure-pipelines/test_plan.py:134]: I'll remove it in subsequent PRs.

**Pull request merged**